dashboard/app/views/levels/levelGroup.js

```diff
@@ -8,19 +8,131 @@ window.getResult = function()
   //   [{"level_id":1977,"result":"0"},{"level_id":2007,"result":"3"},{"level_id":1939,"result":"2,1"}]
   //
 
-  var attemptArray = [];
+  // Add any new results to the existing lastAttempt results.
   for (var i = 0; i < levelCount; i++)
   {
     var multiName = "multi_" + i;
     var multiResult = window[multiName].getCurrentAnswer().toString();
     var levelId = window[multiName].getLevelId();
-    attemptArray.push({level_id: levelId, result: multiResult});
+
+    // But before storing, if we had a previous result for the same level,
+    // remove that from the array, since we want to overwrite that previous
+    // answer.
+    for (var j = lastAttempt.length - 1; j >= 0; j--)
+    {
+      if (lastAttempt[j].level_id == levelId)
+      {
+        lastAttempt.splice(j, 1);
+      }
+    }
+
+    lastAttempt.push({level_id: levelId, result: multiResult});
+  }
+
+  var response = JSON.stringify(lastAttempt);
+
+
+  var forceSubmittable = window.location.search.indexOf("force_submittable") !== -1;
+
+  var result;
+  var submitted;
+
+  if (window.appOptions.level.submittable || this.forceSubmittable)
+  {
+    result = true;
+    submitted = true;
+  }
+  else
+  {
+    result = true; // this.validateAnswers();
+    submitted = false;
   }
-  var response = JSON.stringify(attemptArray);
 
   return {
     "response": response,
     "result": true,
-    "errorType": null
+    "errorType": null,
+    "submitted": submitted
   };
 };
+
+$(document).ready(function() {
+
+  // Are we read-only?  This can be because we're a teacher OR because an answer
+  // has been previously submitted.
+  if (window.appOptions.readonlyWorkspace)
+  {
+    // hide the Submit button.
+    $('.submitButton').hide();
+
+    // Are we a student viewing their own previously-submitted work?
+    if (window.appOptions.submitted)
+    {
+      // show the Unsubmit button.
+      $('.unsubmitButton').show();
```

Comment:
do we show the unsubmit button if we are a teacher AND the answer has been submitted?

I could go either way -- there's an unsubmit button in the teacher panel that is enabled in this case, by design (to 'return' a submission that's wrong and needs to be revised, eg.) so it's not exposing functionality that we don't want to, but I think "student area" stuff is generally all in "read only" mode. We could also show the same button that the student sees and disable it, however I'm not sure if the disabled state of the button is visually distinct enough for that to make sense.